test/serverExplorer.test.ts

```diff
@@ -0,0 +1,140 @@
+
+    test('should able to updateServer the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const serviceStub = sinon.stub(serverExplorer.servers, 'get').returns({
+            stateChange
+        });
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            clear: () => {},
+        });
+        sandbox.stub(serverExplorer, 'refresh')
+        serverExplorer.updateServer(stateChange);
+        expect(serviceStub).calledOnce;
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to showOutput of the server ', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns({
+            show: () => {}
+        });
+        serverExplorer.showOutput(serverHandle);
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to getTreeItem of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.serverStatus, 'get').returns({
+            serverHandle
+        });
+        serverExplorer.getTreeItem(serverHandle);
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to getChildren of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.servers, 'values').returns([]);
+        const getChildren = serverExplorer.getChildren(undefined);
+        expect(getChildren).deep.equals(Array.from(clearStub));
+    });
+
+    test('should able to getTreeItem of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(serverExplorer.serverOutputChannels, 'get').returns(undefined);
+        serverExplorer.addServerOutput(ProcessOutput);
+        expect(clearStub).calledOnce;
+    });
+
+    test('should able to refresh of the server', () => {
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const clearStub = sinon.stub(EventEmitter.prototype, 'fire');
+        serverExplorer.refresh(serverHandle);
+        expect(clearStub).calledOnce;
+        expect(clearStub).calledOnceWith(serverHandle);
+    });
+
+    test('should able to addLocation for the server', async () => {
+        const findServerStub = sinon.stub(clientStub, 'findServerBeans').resolves([findServerBeans]);
+        serverExplorer = new ServersViewTreeDataProvider(clientStub);
+        const showOpenDialogStub = sinon.stub(window, 'showOpenDialog').resolves([{fsPath: 'path/path'}]);
+        sinon.stub(window, 'showInputBox').resolves('eap');
+        sinon.stub(clientStub, 'createServerAsync').resolves(status);
+        await serverExplorer.addLocation();
+        expect(findServerStub).calledOnce;
+        expect(showOpenDialogStub).calledOnce;
+    });
+});
```

Comment:
Add another test ' serverExplorer.addLocation should show message if no server detected in provided location'.